typed-racket-lib/typed-racket/typecheck/tc-metafunctions.rkt

```diff
@@ -74,12 +74,7 @@
         (let* ([p (car worklist)]
                [p (resolve derived-atoms p)])
           (match p
-            [(ImpFilter: a c)
```

Comment:
It was only ever introduced by TR in ```let``` expressions, and we did away with that last year, so I figured it was okay to scrub them from the code base